sdk/search/Azure.Search.Documents/apis.txt

```diff
@@ -31,34 +31,34 @@ namespace Azure.Search.Documents
 
         Response<AutocompleteResults> Autocomplete(string, string, AutocompleteOptions, CancellationToken) { }
         Task<Response<AutocompleteResults>> AutocompleteAsync(string, string, AutocompleteOptions, CancellationToken) { }
-        
+
         Response<IndexDocumentsResult> IndexDocuments(IndexDocumentsBatch<SearchDocument>, IndexDocumentsOptions, CancellationToken) { }
         Task<Response<IndexDocumentsResult>> IndexDocumentsAsync(IndexDocumentsBatch<SearchDocument>, IndexDocumentsOptions, CancellationToken) { }
         Response<IndexDocumentsResult> IndexDocuments<T>(IndexDocumentsBatch<T>, IndexDocumentsOptions, CancellationToken) { }
         Task<Response<IndexDocumentsResult>> IndexDocumentsAsync<T>(IndexDocumentsBatch<T>, IndexDocumentsOptions, CancellationToken) { }
     }
 
-    public class FieldBuilder { }
-    // TODO: FieldBuilder's attributes like [Key], [Searchable], etc.
+    public class SearchIndexBuilder { }
+    // TODO: SearchIndexBuilder's attributes like [Key], [Searchable], etc.
 
     public class SearchServiceClient
     {
+        SearchClient GetSearchClient(string) { }
         SearchIndexClient GetSearchIndexClient(string) { }
-
-
-
-
-
+        SearchDataSourceClient GetSearchDataSourceClient() { }
+        SearchIndexerClient GetSearchIndexerClient(string) { }
```

Comment:
Adding it here since it was brought up internally: we also discussed combining these two since they are so closely related - can't really do one without the other. We should, however, then use method names as they are current, e.g. "CreateDataSource", "CreateIndexer", etc.

Reply:
Why does `GetSearchIndexerClient` take the name as a parameter, but `GetSearchDataSourceClient` doesn't? If these clients do all the CRUD operations, why would any of the methods to get the clients need parameters?

Also, can we drop "Search" from the method names, since it's obvious from the context?

Reply:
As far as combining clients, we need to decide whether we're creating sub-clients per resource, or per scenario. If it's per scenario, then I would argue that data sources, indexers, and skillsets all belong together in something like `SearchEnrichmentClient` (@bleroy -- curious what you think). Ditto for indexes and synonym maps, which are closely related.

Reply:
Good catch @brjohnstmsft about the `name` parameter. I didn't notice that. I would rather keep the clients consistent and simple, as sub-clients that still take all the same parameters we do now. Having to get a new client for each named thing (especially in some cases vs. others) doesn't feel as useful. It's also heavier weight.

Reply:
As discussed, we'll initially use one client per resource (no combining).

Reply:
We also discussed we will not take a name in clients' constructors or in these Get<client> methods but pass them or full models to methods as we currently do.